src/App.js

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import styled from 'styled-components';
+import ReactDOM from 'react-dom';
+
+const Container = styled.div`
+  display: block;  
+  width: 50%;
+  height: 650px;
+  background-color:#F7F2E0; 
+  border: 1px solid black;     
+  border-radius:5px; 
+  margin: auto;
+`
+const Secao = styled.section`
+  margin-top: 0px;
+  background-color:#F7F2E0;
+  height: 80%;
+`
+const Botao =styled.button`
+  color: black;
+  background-color: #F4FA58;
+  border-radius: 10px;
+`
+const Form =styled.div`
+  display: flex;
+  flex-direction:row; 
+  justify-content: space-around;
+  height:30px;
+  border-top: black;
+  background-color:#F7F2E0; 
+  
+`
+const InputUsuario =styled.input`
+  width: 20%;
+  border-radius: 10px;
+  background-color:#F2F5A9;
+`
+const InputMensagem =styled.input`
+  width:63%;
+  border-radius: 10px;
+  background-color:#F2F5A9;
+`
+const CaixaDeMensagem =styled.div`
+display: inline-flex;
+flex-direction: column;
+background-color: #F2F5A9;
+border-radius: 10px;
+margin: 10px;
+
+`
+const Usuario =styled.p`  
+  font-weight: bold;
+  padding: 5px; 
+  margin: 1px;
+`
+const Mensagem =styled.p`
+  padding: 5px;
+  padding-top: 2px;   
+  margin: 1px;
+`
+const Perfil =styled.div`
+display: flex;
+justify-content: flex-end;
+`
+const Titulo= styled.h1`
+text-align:center;
+color:pink;
+`
+class App extends React.Component {
+ 
+  state ={
+    dadosDaMensagem:[
+      {
+        nomeUsuario: "",
+        mensagemUsuario: ""
+      },
+      {
+        nomeUsuario:"",
+        mensagemUsuario:"",
+      }
+    ],
+    valorInputUsuario:"",
+    valorInputMensagem:"",
+  };
+
+  enviaMensagem = (event) => {      
+    const mensagem={      
+      usuario: this.state.valorInputUsuario,      
+      mensagem: this.state.valorInputMensagem
+    };
+
+    console.log("teste " + mensagem.usuario + mensagem.mensagem)//
+
+
+    const novaMensagem = [...this.state.dadosDaMensagem, mensagem];    
+    this.setState({ dadosDaMensagem: novaMensagem}); 
+    this.setState({valorInputMensagem:""});
```

Comment:
Aqui, vcs poderiam usar apenas um this.seState() para mudar todas as propriedades do estado:

```
this.setState({ 
  dadosDaMensagem: novaMensagem,
  valorInputMensagens: "",
})
```